python_modules/libraries/dagster-dbt/dagster_dbt_tests/core/test_asset_checks.py

```diff
@@ -442,3 +444,42 @@ def my_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource):
     assert result.success
     assert len(result.get_asset_materialization_events()) == 1
     assert len(result.get_asset_check_evaluations()) == 2
+
+
+def _get_subset_selection(assets_def: AssetsDefinition, manifest: Dict[str, Any]):
+    context = build_asset_context().bind(
+        assets_def.op, None, assets_def, None, {"dbt": DbtCliResource}
+    )
+    return get_subset_selection_for_context(
+        context,
+        manifest,
+        "my_select_str",
+        "my_exclude_str",
+        dagster_dbt_translator_with_checks,
+    )
+
+
+def test_get_subset_selection_for_context(test_asset_checks_manifest: Dict[str, Any]):
```

Comment:
I was hoping to do some testing directly on `get_subset_selection_for_context` here, but the context from `build_asset_context` doesn't support subsetting so I'd need to construct a proper context from an execution plan

Reply:
I'd prefer testing against the execution of the `@dbt_assets` themselves so we emulate how the user would run into these problems in the first place.

You could do something like:

```python
    @dbt_assets(
        manifest=test_asset_checks_manifest,
        dagster_dbt_translator=dagster_dbt_translator_with_checks,
    )
    def my_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource): 
        dbt_invocation = dbt.cli(["build"], context=context)
        
        # assert transformed CLI arguments from subsetting
        assert dbt_invocation.process.args == ...

        yield from dbt_invocation.stream()
        
   # execute my_dbt_assets
   ...
```